pymongo_auth_aws/auth.py

```diff
@@ -34,18 +37,51 @@
 _AWS_EC2_PATH = 'latest/meta-data/iam/security-credentials/'
 _AWS_HTTP_TIMEOUT = 10
 
-
-AwsCredential = namedtuple('AwsCredential', ['username', 'password', 'token'])
+AwsCredential = namedtuple('AwsCredential', ['username', 'password', 'token', 'expiration'])
 """MONGODB-AWS credentials."""
 
+_cached_credentials = None
+_credential_buffer_seconds = 60 * 5
+
+
+ZERO = timedelta(0)
+
+# A Python 2.7-compliant UTC class (from stdlib docs).
+# When we drop Python 2.7 support we can use `timezone.utc` instead.
+
+class _UTC(tzinfo):
+    """UTC"""
+
+    def utcoffset(self, dt):
+        return ZERO
+
+    def tzname(self, dt):
+        return "UTC"
+
+    def dst(self, dt):
+        return ZERO
+
+utc = _UTC()
+
 
 def _aws_temp_credentials():
     """Construct temporary MONGODB-AWS credentials."""
+    global _cached_credentials
+    creds = _cached_credentials
```

Comment:
A comment would be good here so that future maintainers can understand why they shouldn't try to remove this local variable.

Reply:
Done